### PR TITLE
Added Google Style Docstrings to core.py file methods

### PR DESCRIPTION
**Pull request by adityakaria**
# Description

Added Google Style Docstring to the core.py file of opsdroid to support existing documentation.
Following standards referencing [Sphix 1.8.0+ - Google Style Docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)

Fixes- #532 

## Status
**READY** 

## Type of change

- Documentation (fix or adds documentation)

# How Has This Been Tested?

_No tests Necessary as it is purely documentation_

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: opsdroid/core.py [hunk 1]
```diff
@@ -335,7 +372,17 @@ def _connector_names(self):  # noqa: D401
...
-        """Execute a skill."""
+        """Execute a skill.
+
+        Attempts to run the skill parsed and provides other arguments to the skill if necessary.
+        Also handles the exception encountered if th e
+
+        Args:
+            skill: name of the skill to be run.
+            config: The configuration the skill must be loaded in.
+            message: Message to be parsed to the chat service.
```
Comment (adityakaria) [inline opsdroid/core.py]: @jacobtomlinson I'm not so sure about the role of the message argument. Some guidance on this would be most appreciated!

Comment (jacobtomlinson) [reply opsdroid/core.py]: The `message` argument here will be passed to the skill that is being run. We should probably rename `message` to `event` as we have expanded opsdroid to handle more than just text-based messages.

Comment (adityakaria) [reply opsdroid/core.py]: Ahh, I get it now, thanks! Do you want me to rename it here?

Comment (jacobtomlinson) [reply opsdroid/core.py]: Sure!

**Comment by adityakaria** [comment]
Apologies for the many commits. This is my first time using CI, and so there's some commits I had to revert. But I have not made sure I am not changing any other file apart from the one I'm supposed to.

File: opsdroid/core.py (hunk 1 above)
Comment (adityakaria) [reply opsdroid/core.py]: I have made the changes as advised by you. Kindly take a look, thanks!

**Comment by jacobtomlinson** [comment]
This is great thanks. Don't worry about all the commits, you seem to have learned from raising this PR which is great. We squash on merge anyway.

**Comment by adityakaria** [comment]
Thanks a ton for all your help. Yes, I did learn some new testing and CI methods. 
Looking forward to contributing to Opsdriod again!

**Pull request merged**